packages/common/src/proto-utils.ts

```diff
@@ -0,0 +1,103 @@
+import proto from '@temporalio/proto';
+import { fromProto3JSON, toProto3JSON } from 'proto3-json-serializer';
+import { patchProtobufRoot } from '@temporalio/proto/lib/patch-protobuf-root';
+
+export type History = proto.temporal.api.history.v1.IHistory;
+export type Payload = proto.temporal.api.common.v1.IPayload;
+
+// Cast to any because the generated proto module types are missing the lookupType method
+const patched = patchProtobufRoot(proto) as any;
+const historyType = patched.lookupType('temporal.api.history.v1.History');
+const payloadType = patched.lookupType('temporal.api.common.v1.Payload');
+
+function pascalCaseToConstantCase(s: string) {
+  return s.replace(/[^\b][A-Z]/g, (m) => `${m[0]}_${m[1]}`).toUpperCase();
+}
+
+function fixEnumValue<O extends Record<string, any>>(obj: O, attr: keyof O, prefix: string) {
+  return (
+    obj[attr] && {
+      [attr]: `${prefix}_${pascalCaseToConstantCase(obj[attr])}`,
+    }
+  );
+}
+
+function fixHistoryEvent(e: Record<string, any>) {
+  const type = Object.keys(e).find((k) => k.endsWith('EventAttributes'));
+  if (!type) {
+    throw new TypeError(`Missing attributes in history event: ${JSON.stringify(e)}`);
+  }
+
+  return {
+    ...e,
+    eventType: `EVENT_TYPE_${pascalCaseToConstantCase(e.eventType)}`,
```

Comment:
Nit, seems for consistency could use `...fixEnumValue(e, 'eventType', 'EVENT_TYPE')`